lua/lspinstall/servers/terraform.lua

```diff
@@ -0,0 +1,45 @@
+local config = require'lspconfig'.terraformls.document_config
+require'lspconfig/configs'.terraformls = nil -- important, immediately unset the loaded config again
+config.default_config.cmd[1] = "./terraform-ls"
+
+return vim.tbl_extend('error', config, {
+  install_script = [[
+  os=$(uname -s | tr "[:upper:]" "[:lower:]")
+  arch=$(uname -m | tr "[:upper:]" "[:lower:]")
+
+  case $os in
+  linux)
+    platform="linux"
+  ;;
+  darwin)
+    platform="mac"
+  ;;
+  esac
+
+  case $arch in 
+  i386)
+    architecture=386
+  ;;
+  i586)
+    architecture=386
+  ;;
+  i686)
+    architecture=386
+  ;;
+  x86_64)
+    architecture=amd64
+  ;;
+  arm64)
+    architecture=arm64
+  ;;
+  aarch64)
+    architecture=arm64
+  ;;
+  esac
+  export PATTERN="browser_download_url.*$platform.*$architecture.*"
+  DOWNLOAD_URL=$(curl -s -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/hashicorp/terraform-ls/releases/latest | awk 'BEGIN { FS = "\"" }; $0 ~ ENVIRON["PATTERN"] { print $4 }')
```

Comment:
Why do we need `-H "Accept: application/vnd.github.v3+json"`?

Also, instead of awk and pattern matching, for simplicity I would prefer
` curl -s https://api.github.com/repos/hashicorp/terraform-ls/releases/latest | grep 'browser_' | cut -d\" -f4 | grep "$platform" | grep "$architecture"`

Reply:
Just following the Github docs for Accept: https://docs.github.com/en/rest/overview/media-types. I will try remove and retest.

Reply:
Works just as well without the Accept so will remove it.

Reply:
Have switched to the cut and grep combo.